src/core/utils/channel_manager.cpp

```diff
@@ -98,18 +99,42 @@ void ChannelManager::PreparePendingDataset(void)
 {
     ThreadNetif &netif = GetInstance().GetThreadNetif();
     uint64_t pendingTimestamp = 0;
+    uint64_t pendingActiveTimestamp = 0;
+    uint32_t delayInMs = TimerMilli::SecToMsec(static_cast<uint32_t>(mDelay));
     otOperationalDataset dataset;
-    uint32_t delayInMs;
     otError error;
 
     VerifyOrExit(mState == kStateChangeRequested);
 
+    VerifyOrExit(mChannel != GetInstance().Get<Mac::Mac>().GetChannel());
+
+    if ((mSupportedChannels & (1U << mChannel)) == 0)
+    {
+        otLogInfoUtil(GetInstance(), "ChannelManager: Request rejected! Channel %d not in supported mask 0x%x",
```

Comment:
Should there be an `ExitNow()` here?

Reply:
Indeed! Thanks.  (also need to set the `mState`)...